test/test_plugin_helper.rb

```diff
@@ -6,3 +6,21 @@
 # Add plugin to FactoryGirl's paths
 FactoryGirl.definition_file_paths << File.join(File.dirname(__FILE__), 'factories')
 FactoryGirl.reload
+
+# Foreman's setup doesn't handle cleaning up for Minitest::Spec
+DatabaseCleaner.strategy = :transaction
+
+class Minitest::Spec
+  class << self
+    alias_method :context, :describe
+  end
+
+  before :each do
+    DatabaseCleaner.start
+    Setting::RemoteExecution.load_defaults
+  end
+
+  after :each do
+    DatabaseCleaner.clean
```

Comment:
It is odd, and you won't see it on jenkins because you start with a fresh DB.  Feel free to run them locally twice, and you'll see objects created during tests aren't cleaned out of the database without this.  At least that's what it does for me on my katello-dev box which uses postgresql.